src/content/components/NodeIcon.js

```diff
@@ -0,0 +1,35 @@
+import React, { PureComponent, PropTypes } from 'react';
+import { connect } from 'react-redux';
+
+import { getIconClassNameForNode } from '../reducers/icons';
+import actions from '../actions';
+
+class NodeIcon extends PureComponent {
+  constructor(props) {
+    super(props);
+    if (props.icon) {
+      props.iconStartLoading(props.icon);
+    }
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.icon) {
+      nextProps.iconStartLoading(nextProps.icon);
+    }
+  }
+
+  render() {
+    return <div className={`treeRowIcon ${this.props.className || ''}`}></div>;
```

Comment:
@mstange should we be using the [classname](https://www.npmjs.com/package/classnames) module for stuff like this?